### ISSUE TITLE
Possibly incorrect implementations of `leading_ones` and `leading_zeros`

### ISSUE THREAD

**Issue by mcncm** [bitvecto-rs/bitvec#127]
From `BitSlice` (`leading_zeros` is symmetric),
```
pub fn leading_ones(&self) -> usize {
	self.first_zero().unwrap_or_default()
}
```	
This function fails the test `assert_eq!(bits![1, 1, 1].leading_ones(), 3);`, which is not listed in the doc comment. Is this behavior intentional? If not, an implementation such as 
```
pub fn leading_ones(&self) -> usize {
	self.first_zero().unwrap_or_else(|| self.len())
}
```
would fix it (though may or may not be the most efficient). If the behavior is intentional, it may be surprising to some users, and could stand to be documented.

Thanks for reading.

**Comment by myrrlyn** [comment]
This was reported and fixed in #120 but I don't think I've published it to any of the released series. I apologize.

**Comment by mcncm** [comment]
Ah no, my apologies--I should have taken a look at the closed issues. Thanks for the reply!
